lib/src/navbar_router.dart

```diff
@@ -359,10 +410,25 @@ class _NavbarRouterState extends State<NavbarRouter>
                     /// same duration as [_AnimatedNavbar]'s animation duration
                     duration: const Duration(milliseconds: 500),
                     padding: EdgeInsets.only(left: getPadding()),
-                    child: Stack(children: [
-                      for (int i = 0; i < NavbarNotifier.length; i++)
-                        _buildIndexedStackItem(i, context)
-                    ]),
+                    child: ListView.builder(
+                        itemCount: NavbarNotifier.length,
+                        scrollDirection: Axis.horizontal,
+                        physics: const NeverScrollableScrollPhysics(),
+                        controller: pageGesture.pageController,
+                        itemBuilder: (context, i) {
+                          // use keep-alive to prevent list builder to rebuild
+                          return KeepAliveWrapper(
```

Comment:
When we horizontally swipe the widget should be visible even if the drag did not complete, currently I see a blank white screen until the drag completes

for instance if you are on index 0 and make start dragLeft and drag lets say 60%  without leaving your touch on the screen, I see the tab on index 2 has not yet rendered or visible it should always be visible as you start scrolling.

To better understand this I highly recommend trying out the swipe gesture on Whatsapp app for Android.

Reply:
I did, and I recommend you to take a look at the demo video and see it had been implemented in the first place before you told me to refractor the code somehow.